wire-ios-data-model/Source/Model/Message/ZMAssetClientMessage+Ephemeral.swift

```diff
@@ -65,15 +65,26 @@ extension ZMAssetClientMessage {
     }
 
     @discardableResult @objc public override func startDestructionIfNeeded() -> Bool {
-
-        if imageMessageData != nil && !hasDownloadedFile {
-            return false
-        } else if fileMessageData != nil
-            && underlyingMessage?.assetData?.hasUploaded == false
-            && underlyingMessage?.assetData?.hasNotUploaded == false {
-            return false
+        if let isSelfUser = sender?.isSelfUser {
+            // check for download state only for images that were sent by another user
+            if !isSelfUser && managedObjectContext?.zm_isUserInterfaceContext == true
+                && imageMessageData != nil
+                && !hasDownloadedFile {
+                return false
+            } else if fileMessageData != nil
+                        && underlyingMessage?.assetData?.hasUploaded == false
+                        && underlyingMessage?.assetData?.hasNotUploaded == false {
+                    return false
+            }
+        } else {
+            if imageMessageData != nil && !hasDownloadedFile {
+                return false
+            } else if fileMessageData != nil
+                && underlyingMessage?.assetData?.hasUploaded == false
+                && underlyingMessage?.assetData?.hasNotUploaded == false {
+                return false
+            }
```

Comment:
I wonder if we can improve this if else return false...

Like store the conditions to more meaningful constant name and if/else at the end on them